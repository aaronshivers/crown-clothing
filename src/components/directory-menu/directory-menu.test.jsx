import React from 'react';
import { shallow } from 'enzyme';
import DirectoryMenu from './directory-menu.component';

describe('DirectoryMenu', () => {
  const directoryMenu = shallow(
    <DirectoryMenu />,
  );

  it('should render correctly', () => {
    expect(directoryMenu).toMatchSnapshot();
  });

  it('should have a class of `directory-menu`', () => {
    expect(directoryMenu.hasClass('directory-menu')).toBeTruthy();
  });

  it('should contain 5 `MenuItem` components', () => {
    expect(directoryMenu.find('MenuItem')).toHaveLength(5);
  });

  it('should render the correct title text', () => {
    expect(directoryMenu.find('MenuItem').get(0).props.title).toBe('hats');
    expect(directoryMenu.find('MenuItem').get(1).props.title).toBe('jackets');
    expect(directoryMenu.find('MenuItem').get(2).props.title).toBe('sneakers');
    expect(directoryMenu.find('MenuItem').get(3).props.title).toBe('womens');
    expect(directoryMenu.find('MenuItem').get(4).props.title).toBe('mens');
  });
});
