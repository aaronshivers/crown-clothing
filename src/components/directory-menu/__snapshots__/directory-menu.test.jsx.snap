// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectoryMenu should render correctly 1`] = `
<div
  className="directory-menu"
>
  <MenuItem
    imageUrl="https://i.ibb.co/cvpntL1/hats.png"
    key="1"
    title="hats"
  />
  <MenuItem
    imageUrl="https://i.ibb.co/px2tCc3/jackets.png"
    key="2"
    title="jackets"
  />
  <MenuItem
    imageUrl="https://i.ibb.co/0jqHpnp/sneakers.png"
    key="3"
    title="sneakers"
  />
  <MenuItem
    imageUrl="https://i.ibb.co/GCCdy8t/womens.png"
    key="4"
    size="large"
    title="womens"
  />
  <MenuItem
    imageUrl="https://i.ibb.co/R70vBrQ/men.png"
    key="5"
    size="large"
    title="mens"
  />
</div>
`;
